src/Features/LanguageServer/Protocol/RoslynRequestExecutionQueue.cs

```diff
@@ -39,6 +45,55 @@ public override Task WrapStartRequestTaskAsync(Task nonMutatingRequestTask, bool
             }
         }
 
+        /// <inheritdoc/>
+        protected override IMethodHandler GetHandlerForRequest(IQueueItem<RequestContext> work)
+        {
+            var defaultHandlerMetadata = new RequestHandlerMetadata(work.MethodName, work.RequestType, work.ResponseType, LanguageServerConstants.DefaultLanguageName);
+            var defaultHandler = _handlerCache.GetOrAdd(defaultHandlerMetadata, metadata => _handlerProvider.GetMethodHandler(metadata.MethodName, metadata.RequestType, metadata.ResponseType, metadata.Language));
```

Comment:
I don't like having an additional cache here - handlerProvider should be solely responsible for caching and fast pathing requests for handlers.